src/components/Views/MapView/Map/Map.js

```diff
@@ -110,13 +110,64 @@ const mapboxStyles = MapboxGL.StyleSheet.create({
   }
 });
 
+const emptyFeatureCollection = {
+  type: 'FeatureCollection',
+  features: []
+};
+
+type ObsMapLayerProps = {
+  onPress: Function,
+  observations: {
+    [id: string]: ObservationType
+  }
+};
+
+function mapObservationsToFeatures(obs: { [string]: ObservationType }) {
+  if (!obs || isEmpty(obs)) return [];
+  return Object.keys(obs)
+    .filter(
+      id => typeof obs[id].lon !== undefined && typeof obs[id].lat !== undefined
+    )
+    .map(id => ({
+      type: 'Feature',
+      geometry: {
+        type: 'Point',
+        coordinates: [obs[id].lon, obs[id].lat]
+      },
+      properties: {
+        id: id,
+        categoryId: obs[id].categoryId
+      }
+    }));
+}
+
+class ObservationMapLayer extends React.PureComponent<ObsMapLayerProps> {
+  render() {
+    const { onPress, observations } = this.props;
+    const featureCollection = {
+      type: 'FeatureCollection',
+      features: mapObservationsToFeatures(observations)
+    };
+    return (
+      <MapboxGL.ShapeSource
+        onPress={onPress}
+        key={`observations-source`}
+        id={`observations-source`}
+        shape={featureCollection}
+      >
+        <MapboxGL.CircleLayer id={`circles`} style={mapboxStyles.observation} />
+      </MapboxGL.ShapeSource>
+    );
+  }
+}
+
 class Map extends React.Component<Props & StateProps & DispatchProps> {
   map: any;
 
   constructor() {
     super();
-
     MapboxGL.setAccessToken(env.accessToken);
+    this.handleObservationPress = this.handleObservationPress.bind(this)
```

Comment:
The fancy syntax `handleObservationPress = (e) => {` is shorthand for this (as opposed to normal class methods written as `handleObservationPress (e) {`) so this is not needed.

Reply:
nice catch!